### PR TITLE
Add HCI server workbook template

### PR DESCRIPTION
**Pull request by myunghojung**
## PR Checklist

* [ *] Explain your changes, so people looking at the PR know *what* and *why*, the code changes are the *how*.
This change adds a template for Azure Stack HCI Server workbook. The dashboard shows performance, health and usage of a registered Azure Stack HCI server that are connected to Azure and are enrolled in monitoring.

image
image
image

### If adding or updating templates:
* [*] post a screenshot of templates and/or gallery changes
* [*] ensure your template has a corresponding gallery entry in the gallery folder
* [*] If you are adding a new template, add your team and template/gallery file(s) to the CODEOWNERS file. CODEOWNERS entries should be teams, not individuals
* [*] ensure all steps have meaningful names
* [*] ensure all parameters and grid columns have display names set so they can be localized
* [*] ensure that parameters id values are unique __or they will fail PR validation__ (parameter ids are used for localization)
* [*] ensure that steps names are unique __or they will fail PR validation__ (step names are used for localization)
* [*] grep `/subscription/` and ensure that your parameters don't have any hardcoded resourceIds __or they will fail PR validation__
* [*] remove `fallbackResourceIds` and `fromTemplateId` fields from your template workbook __or they will fail PR validation__

### CODE REVIEW

File: Workbooks/Azure Stack HCI/ServerUsage/ServerUsage.workbook [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+            "name": "resource",
+            "type": 5,
+            "isRequired": true,
+            "query": "where type =~ 'microsoft.azurestackhci/clusters'\r\n| where id =~ '{resourceId}'",
+            "isHiddenWhenLocked": true,
+            "typeSettings": {
+              "resourceTypeFilter": {
+                "microsoft.azurestackhci/clusters": true
+              },
+              "nameFilter": "'n26-20-01-c3'",
```
Comment (yves-chan) [inline Workbooks/Azure Stack HCI/ServerUsage/ServerUsage.workbook:82]: is this part needed?

File: Workbooks/Azure Stack HCI/ServerUsage/ServerUsage.workbook [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+                    "version": "KqlItem/1.0",
+                    "query": "let ArmId = Event\r\n| where EventLog =~ \"Microsoft-Windows-SDDC-Management/Operational\" and EventID == \"3000\" \r\n| where TimeGenerated {timeRange:query}\r\n| extend ClsuterData = parse_xml(EventData)\r\n| extend ClusterName = tostring(ClsuterData.DataItem.UserData.EventData[\"ClusterName\"])\r\n| extend ClusterArmId = tostring(ClsuterData.DataItem.UserData.EventData[\"ArmId\"])\r\n| where ClusterArmId =~ '{resourceId}'\r\n| summarize arg_max(TimeGenerated,*) by ClusterArmId\r\n| extend ArmPrefix = extract(@'(.+/)[^/]+$', 1, _ResourceId)\r\n| extend servers_information = parse_json(RenderedDescription).m_servers\r\n| mv-expand servers_information\r\n| extend Nodename = tostring(servers_information.m_name)\r\n| where Nodename =~ '{server}'\r\n| project _ResourceId = tolower(strcat(ArmPrefix, Nodename));\r\n\r\nPerf\r\n| where ObjectName == \"RDMA Activity\"\r\n| summarize Computer = any(Computer), _ResourceId = any(_ResourceId), RdmaUsage = sum(CounterValue) by TimeGenerated\r\n| join kind=leftsemi ArmId on _ResourceId\r\n| project NodeName = split(Computer, \".\")[0], RdmaUsage, TimeGenerated",
```
Comment (yves-chan) [inline Workbooks/Azure Stack HCI/ServerUsage/ServerUsage.workbook:290]: small typo on ClsuterData

File: Workbooks/Azure Stack HCI/ServerUsage/ServerUsage.workbook (hunk 1 above)
Comment (myunghojung) [reply Workbooks/Azure Stack HCI/ServerUsage/ServerUsage.workbook:82]: This workbook will be opened from link in another workbook. So, the parameter is passed from the workbook and this parameter converting ARM ID from text to resource type. It makes easier to filter query by subscription ID in ARM ID. I'm testing and haven't find any better way to extract subscription ID and filter following queries with it yet. I'll remove or replace this if I find better solution. Please let me know if there is.

File: Workbooks/Azure Stack HCI/ServerUsage/ServerUsage.workbook (hunk 2 above)
Comment (myunghojung) [reply Workbooks/Azure Stack HCI/ServerUsage/ServerUsage.workbook:290]: Good catch, thank you!

**Review by muthus-msft — APPROVED** [review]

**Pull request merged**